ui/apps/dashboard/src/components/ArchivedFuncBanner.tsx

```diff
@@ -0,0 +1,51 @@
+'use client';
+
+import { useEnvironment } from '@/components/Environments/environment-context';
+import { graphql } from '@/gql';
+import { useGraphQLQuery } from '@/utils/useGraphQLQuery';
+import { Banner } from './Banner';
+
+const Query = graphql(`
+  query GetArchivedFuncBannerData($envID: ID!, $funcID: ID!) {
+    environment: workspace(id: $envID) {
+      function: workflow(id: $funcID) {
+        id
+        archivedAt
+      }
+    }
+  }
+`);
+
+type Props = {
+  funcID: string;
+};
+
+export function ArchivedFuncBanner({ funcID }: Props) {
+  const env = useEnvironment();
+  const { data, error } = useGraphQLQuery({
+    query: Query,
+    variables: {
+      envID: env.id,
+      funcID,
+    },
+  });
+  if (error) {
+    console.error(error);
+    return null;
+  }
+
+  if (!data?.environment.function?.archivedAt) {
+    return null;
+  }
+
+  if (env.isArchived) {
+    // We don't want both this banner and the env banner
+    return null;
+  }
+
+  return (
+    <Banner kind="warning">
+      <span className="font-semibold">Function is archived.</span>
```

Comment:
Can we also mention that the function can be archived by adding it back to the app and resyncing?

Reply:
And maybe mention that it was archived because its app was synced without it

Reply:
✅ Done in dcb7f828980fb8fd0941462f82636a4ebbd51479